[Catherine Liang] iP

ketweeen opened the pull request:
## **ValeriePro** ♡

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


ValeriePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _FAST SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks (todo, deadline, or event).
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ketweeen committed:
Level-0 changes

ketweeen committed:
Level-1 changes: allowed for user input, if user inputs bye it quits program and else becomes added: (user input)

ketweeen committed:
Added Level-2 changes: user inputs will now added to a list, inputting list will print out the list

ketweeen committed:
Level-3 changes: Added Task class, userList now stores Task instances, user can also mark and unmark tasks.

ketweeen committed:
Level-4 changes: Added ToDos, Deadlines, and Events class. User can now make an todos, deadlines, or events task.

ketweeen committed:
A-TextUiTesting changes: changed input.txt and EXPECTED.txt for testing purposes

ketweeen committed:
Level-5 changes: Added error handlings when creating a new task; inputing something other than list, bye, todo, deadline, or event will return an error message

ketweeen committed:
Level-6 changes: User can now delete tasks.

ketweeen committed:
Update Deadlines, Duke, Events, Task, and ToDos

Users can now load and save files.

ketweeen committed:
Merge branch 'branch-Level-7'

ketweeen committed:
Update Deadline, Duke, and Task

The string inputted after /by is now treated as a LocalDate.

ketweeen committed:
Merge branch 'branch-Level-8'

ketweeen committed:
Add Parser, Storage, TaskList, and Ui as well as update Duke and Task

Incorporate more OOP

ketweeen committed:
Merge branch 'A-MoreOOP'

ketweeen committed:
Add duke package to all class and do some bug fixes

"bye" command is moved from Parser class to Duke class so loop exits after bye.

ketweeen committed:
Merge branch 'branch-A-Packages'

ketweeen committed:
Move fromString method to Parser class and change name to parseFileInput

ketweeen committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ketweeen committed:
Add ParserTest and TaskTest

Added JUnit Test methods

ketweeen committed:
Merge branch 'A-JUnit'

ketweeen committed:
Update Duke, Parser, Storage, Task, TaskList, Ui, ParserTest, and TaskTest

Added JavaDocs headers to the methods.

ketweeen committed:
Merge branch 'branch-A-JavaDoc'

ketweeen committed:
Update Parser

Changed the indentation of case (of a switch)

ketweeen committed:
Merge branch 'branch-A-CodingStandard'

ketweeen committed:
Update Parser, TaskList, Ui

Users can now find tasks.

ketweeen committed:
Merge branch 'branch-Level-9'

ketweeen committed:
Update TaskList and Ui

Added JavaDoc headers to new methods

ketweeen committed:
Merge branch 'branch-Level-9'

jonasongg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,195 @@
+            String[] parts = userInput.split(" ");
+            if (parts.length == 2) {
+                try {
+                    int index = Integer.parseInt(parts[1]) - 1;
```
Might be better to reduce the code nesting in this method? Perhaps `return;` in the first `if` to remove the `else if`

jonasongg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,195 @@
+                if (userInput.length() < 5) {
+                    // Incorrect input format for todos
+                    Ui.showError("Incorrect input format for todo");
+                } else {
```
Same as above; might be better to reduce code nesting here as well?

jonasongg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,195 @@
+            break;
+        case "D":
+            String by = parts[3].trim();
+
```
I like how you separated these logically separate statements by blank lines!

jonasongg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,195 @@
+
+            task = new Deadlines(description, deadlineDate);
+            break;
+        case "E":
```
Might be better to add a default branch to ensure all bases are covered? If control flow really should not reach the default branch, it might be better to raise an exception to detect any errors in the program!

ketweeen committed:
Update files according to checkstyle

ketweeen committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	src/main/java/duke/Parser.java

ketweeen committed:
Update Parser

owenyeo commented on src/main/java/duke/Deadlines.java:
I like how clean your code looks! Keep it up.

owenyeo commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,41 @@
+/**
+ * A class for Deadlines tasks.
+ */
+public class Deadlines extends Task {
```
Perhaps the name of this class can be named "Deadline" instead? Since an instance of "Deadlines" is referring to a singular deadline.

owenyeo commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,41 @@
+/**
+ * A class for Events tasks.
+ */
+public class Events extends Task {
```
Similar to Deadlines!

owenyeo commented on src/main/java/duke/ToDos.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * A class for ToDos tasks.
+ */
+public class ToDos extends Task {
```
Same comment as Deadlines and Events

owenyeo reviewed (commented):
Your coding style is very clean looking, and great use of the appropriate methods in parser! Your JavaDocs are also on point, with each one being very clear and in line with JavaDoc conventions. Perhaps you could think of ways to make it cleaner using Enums, or with increased abstraction?

ketweeen committed:
Integrated JavaFx GUI

Changed file names from Duke to Valerie

ketweeen committed:
Merge branch 'branch-Level-10'

ketweeen committed:
Update build.gradle

When a JAR file is created, it will now be named valerie.jar

ketweeen committed:
Add assert on various classes and minor bug fixes

Some methods lack proper assurance of their correct functionality. The lack of assurance can lead to potential errors or unexpected behavior in the code.

Implement assertions to ensure the methods operate as intended.

Additionally, a minor bug with Valerie's response presentation has been fixed. Now, all responses are displayed in a single dialog box for improved user experience.

ketweeen committed:
Merge branch 'branch-A-Assertions'

ketweeen committed:
Update readibility of methods in MainWindow and Parser class

Code duplication exists in MainWindow, and the parseUserInput() method is excessively long. Code duplication can lead to maintenance challenges and increased chances of introducing bugs. Lengthy methods can be harder to understand and maintain.

This change aims to enhance the codebase's maintainability and readability, contributing to a more efficient development process.

ketweeen committed:
Merge branch 'branch-A-CodeQuality'

ketweeen committed:
Add 'help' command

The system lacks a 'help' command for users to access a list of available commands.

Adding a 'help' command enhances the user experience by providing guidance and information about available commands.  Users will also be notified about this command both at launch and when entering an unknown command.

ketweeen committed:
Merge branch 'branch-C-Help'

ketweeen committed:
Add 'assert' on Task class

ketweeen committed:
Merge pull request #1 from ketweeen/branch-A-Assertions

Add 'assert' on Task class

ketweeen committed:
Merge branch 'master' of https://github.com/ketweeen/ip

ketweeen committed:
Remove magic numbers from Parser class

ketweeen committed:
Merge branch 'master' into branch-A-CodeQuality

ketweeen committed:
Merge pull request #2 from ketweeen/branch-A-CodeQuality

Remove magic numbers from Parser class

ketweeen committed:
Update JavaDocs and fix checkstyle issues

ketweeen committed:
Update README file

ketweeen committed:
Merge branch 'branch-A-UserGuide'

ketweeen committed:
Update README

ketweeen committed:
Update TaskList, Ui, and Valerie

The chat bot will now closes once the user inputs "bye". The chat bot will also return an error if a user does a list command when the task list is empty.
